### ISSUE TITLE
$config['port'] somehow gets set to tcp://172.17.0.27:5672

### ISSUE THREAD

**Issue by jessemortenson** [vladimir-yuldashev/laravel-queue-rabbitmq#39]
I've encountered this very strange behavior where $config['port'] = 'tcp://172.17.0.27:5672' inside of  RabbitMQConnector->connect() (line 22). I would expect $config['port'] to equal simply 5672. (172.17.0.27 is the resolved IP of my docker container). As a result of this behavior, the connection to RabbitMQ fails (new AMQPConnection($config['host'], $config['port'] ...) and I can't consume messages via the artisan command (php artisan queue:listen).

I can't figure out how $config['port'] gets transformed into a full protocol://hostname:port string. In my bootstrap/app.php, I check $app['config']["queue.connections"] right after service providers and configuration is registered, and it looks the way I expect it to. So it *seems* like my configuration files are correct.

I am using the library inside of Lumen, so maybe that's a difference. PHP 5.6.7. My composer.json line is: "fintech-fab/laravel-queue-rabbitmq": "5.1". Running a rabbitmq 3.5.5 server.

**Comment by vladimir-yuldashev** [comment]
Show your configuration file, please

**Comment by jessemortenson** [comment]
config/queue.php
```
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Queue Driver
    |--------------------------------------------------------------------------
    |
    | The Laravel queue API supports a variety of back-ends via an unified
    | API, giving you convenient access to each back-end using the same
    | syntax for each one. Here you may set the default queue driver.
    |
    | Supported: "null", "sync", "database", "beanstalkd",
    |            "sqs", "iron", "redis"
    |
    */

    'default' => env('QUEUE_DRIVER', 'rabbitmq'),

    /*
    |--------------------------------------------------------------------------
    | Queue Connections
    |--------------------------------------------------------------------------
    |
    | Here you may configure the connection information for each server that
    | is used by your application. A default configuration has been added
    | for each back-end shipped with Laravel. You are free to add more.
    |
    */

    'connections' => [

        'rabbitmq' => [
            'driver' => 'rabbitmq',
            'host' => env('RABBITMQ_HOST', 'rabbitmq'),
            'port' => env('RABBITMQ_PORT', 5672),
            'vhost' => env('RABBITMQ_VHOST', '/'),
            'login' => env('RABBITMQ_LOGIN', 'admin'),
            'password' => env('RABBITMQ_PASSWORD', 'password'),

            'queue' => env('RABBITMQ_QUEUE'), // name of the default queue,

            // create the exchange if not exists
            'exchange_declare' => env('RABBITMQ_EXCHANGE_DECLARE', true),
            // create the queue if not exists and bind to the exchange
            'queue_declare_bind' => env('RABBITMQ_QUEUE_DECLARE_BIND', true),

            'queue_params' => [
                'passive' => env('RABBITMQ_QUEUE_PASSIVE', false),
                'durable' => env('RABBITMQ_QUEUE_DURABLE', true),
                'exclusive' => env('RABBITMQ_QUEUE_EXCLUSIVE', false),
                'auto_delete' => env('RABBITMQ_QUEUE_AUTODELETE', false),
            ],

            'exchange_params' => [
                'name' => env('RABBITMQ_EXCHANGE_NAME', null),
                // more info at http://www.rabbitmq.com/tutorials/amqp-concepts.html
                'type' => env('RABBITMQ_EXCHANGE_TYPE', 'direct'),
                'passive' => env('RABBITMQ_EXCHANGE_PASSIVE', false),
                'durable' => env('RABBITMQ_EXCHANGE_DURABLE', true), // the exchange will survive server restarts
                'auto_delete' => env('RABBITMQ_EXCHANGE_AUTODELETE', false),
            ],
        ],
        'sync' => [
            'driver' => 'sync',
        ],

        'database' => [
            'driver' => 'database',
            'table' => 'jobs',
            'queue' => 'default',
            'expire' => 60,
        ],

        'beanstalkd' => [
            'driver' => 'beanstalkd',
            'host'   => 'localhost',
            'queue'  => 'default',
            'ttr'    => 60,
        ],

        'sqs' => [
            'driver' => 'sqs',
            'key'    => 'your-public-key',
            'secret' => 'your-secret-key',
            'queue'  => 'your-queue-url',
            'region' => 'us-east-1',
        ],

        'iron' => [
            'driver'  => 'iron',
            'host'    => 'mq-aws-us-east-1.iron.io',
            'token'   => 'your-token',
            'project' => 'your-project-id',
            'queue'   => 'your-queue-name',
            'encrypt' => true,
        ],

        'redis' => [
            'driver' => 'redis',
            'connection' => 'default',
            'queue'  => 'default',
            'expire' => 60,
        ],

    ],

    /*
    |--------------------------------------------------------------------------
    | Failed Queue Jobs
    |--------------------------------------------------------------------------
    |
    | These options configure the behavior of failed queue job logging so you
    | can control which database and table are used to store the jobs that
    | have failed. You may change them to any database / table you wish.
    |
    */

    'failed' => [
        'database' => 'mysql', 'table' => 'failed_jobs',
    ],

];
```

and .env

```
APP_ENV=local
APP_DEBUG=true
APP_KEY=[real-key-value]

APP_LOCALE=en
APP_FALLBACK_LOCALE=en

DB_CONNECTION=mysql
DB_HOST=db
DB_PORT=3306
DB_DATABASE=[real-db-value]
DB_USERNAME=[real-user-value]
DB_PASSWORD=[real-password-value]

DOCTRINE_PROXY_AUTOGENERATE=true

# CACHE_DRIVER=database
# SESSION_DRIVER=database
QUEUE_DRIVER=rabbitmq

RABBITMQ_HOST=rabbitmq
RABBITMQ_PORT=5672
RABBITMQ_VHOST=/
RABBITMQ_LOGIN=admin
RABBITMQ_PASSWORD=[real-password-value]
RABBITMQ_QUEUE=showbill_process_jobs
```

**Comment by vladimir-yuldashev** [comment]
RABBITMQ_HOST=rabbitmq

are you sure it's correct?

**Comment by jessemortenson** [comment]
Yes (I'm using a docker container). 

My setup *is* working, as long as I modify the $config['port'] value used in RabbitMQConnector->connect(), like so:

```
        // JKM HACK somehow $config['port'] is getting turned into protocol://host:port
        $portParts = explode(':', $config['port']);
        if (isset($portParts[2])) {
            $config['port'] = $portParts[2];
        }
```

**Comment by vladimir-yuldashev** [comment]
Did you solve the issue?

**Comment by jessemortenson** [comment]
Sorry! I finally figured this one out. You were right to ignore this issue. Turns out it was caused by a quirk of my docker-compose configuration. The container was named "rabbitmq" and the docker setup was automatically injecting a bunch of variables into $_SERVER, including the ports exposed from various containers. So there was a $_SERVER['RABBITMQ_PORT'] entry set to 'tcp://172.17.0.27:5672' by the Docker setup. 

Total coincidence that this is also the name of the .env key that config is checking for (getenv('RABBITMQ_PORT'], 5672)). Closing issue as "won't fix - unrelated to this project"
